### PR TITLE
[AVIATO-628] [Calendar] Add new cell style and update styles to match designs

### PR DESCRIPTION
**Pull request by fedetrim**
This PR:
- Adds a new cell style `.noData` to allow consumers to indicate a day for which there is no price information available
- Updates the calendar cell styles to match the current designs

|  | Light | Dark |
|-|-|-|
| Default | Simulator Screen Shot - iPhone 11 Pro - 2020-11-05 at 14 25 46 | Simulator Screen Shot - iPhone 11 Pro - 2020-11-05 at 14 25 53 |
| With Price Coloring | Simulator Screen Shot - iPhone 11 Pro - 2020-11-05 at 14 02 42 | Simulator Screen Shot - iPhone 11 Pro - 2020-11-05 at 14 02 47 |

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by fedetrim** [comment]
CI failed with this error:

```
> mdspell -r --en-gb --ignore-acronyms --ignore-numbers --no-suggestions '*.md' '**/*.md' '!**/node_modules/**/*.md' '!Example/Pods/**/*.md' '!vendor/**/*.md' '!gems/**/*.md'

    UNRELEASED.md
        8 | KCalendarDateCellStyleNoData` enum case to indicate a calendar c 

>> 1 spelling error found in 47 files
```

Can you help understand specifically what's the spelling error?

File: Backpack/Calendar/Classes/BPKCalendar.m [hunk 1]
```diff
@@ -629,11 +628,13 @@ + (UIColor *)fillColorForDateStyle:(BPKCalendarDateCellStyle)style
...
         case BPKCalendarDateCellStylePositive:
-            return BPKColor.glencoe;
+            return [BPKColor dynamicColorWithLightVariant:BPKColor.monteverde darkVariant:BPKColor.glencoe];
         case BPKCalendarDateCellStyleNeutral:
             return BPKColor.erfoud;
         case BPKCalendarDateCellStyleNegative:
-            return BPKColor.hillier;
+            return BPKColor.panjin;
+        case BPKCalendarDateCellStyleNoData:
+            return BPKColor.textTertiaryColor;
```
Comment (k0nserv) [inline Backpack/Calendar/Classes/BPKCalendar.m:637]: I'm unsure about using `textTertiary` here, this is a background, not text so it feels semantically incorrect

File: Backpack/Calendar/Classes/BPKCalendar.m [hunk 2]
```diff
@@ -651,9 +652,12 @@ + (UIColor *)titleColorForDateStyle:(BPKCalendarDateCellStyle)style
 
         // HERE BE DRAGONS: Explicit fallthrough
         case BPKCalendarDateCellStylePositive:
-        case BPKCalendarDateCellStyleNeutral:
         case BPKCalendarDateCellStyleNegative:
-            return BPKColor.skyGray;
+        case BPKCalendarDateCellStyleNoData:
+            return BPKColor.backgroundColor;
```
Comment (k0nserv) [inline Backpack/Calendar/Classes/BPKCalendar.m:657]: Similarily to the above this feels semantically wrong

**Comment by shaundon** [comment]
@fedetrim I think it's `enum`, because it's not a word in the dictionary (being short for `enumeration`). Could you add it to `.spelling` and see if that fixes it?

**Comment by k0nserv** [comment]
@fedetrim Can you add a note about the current day no longer having a special style?

**Comment by fedetrim** [comment]
Forgot to update snapshots for that last color change 😅
Doing now

**Comment by fedetrim** [comment]
@k0nserv all done 😄

**Review by k0nserv — APPROVED** [review]
Great, can you squash the commits and I'll merge?

**Comment by fedetrim** [comment]
@k0nserv what about GitHub's `Squash and merge` which can do this automatically?

**Comment by k0nserv** [comment]
> @k0nserv what about GitHub's Squash and merge which can do this automatically?

I like to keep the merge commits that GitHub generates

**Pull request merged**